teuthology/suite/test/test_run_.py

```diff
@@ -360,3 +360,52 @@ def test_newest_success(
         m_find_git_parents.assert_has_calls(
             [call('ceph', 'ceph_sha1', 10)]
         )
+
+    def test_dupa(
```

Comment:
very funny, I guess you've used Polish native-speaking skills here ;-) may we have some more appropriate name, like: `test_<func-name-we-test>`?

Reply:
I mean:
```
    @pytest.mark.parametrize('desc, base, result', [
        [
            "rados/objectstore/{backends/objectstore-bluestore-b supported-random-distro$/{ubuntu_latest}}",
            "/cephfs/github.com_ceph_build/qa/suites",
            [
               ('rados/objectstore/{backends/objectstore-bluestore-b supported-random-distro$/{ubuntu_latest}}',
                ['/cephfs/github.com_ceph_build/qa/suites/rados/objectstore/backends/objectstore-bluestore-b.yaml',
                 '/cephfs/github.com_ceph_build/qa/suites/rados/objectstore/supported-random-distro$/ubuntu_latest.yaml'])
            ]
        ],
        [
            "rados/singleton-nomsgr/{all/health-warnings mon_election/connectivity rados supported-random-distro$/{ubuntu_latest}}",
            "/cephfs/home/user/src/git.ceph.com_ceph-c_aclamk-testing/qa/suites",
            [
                ("rados/singleton-nomsgr/{all/health-warnings mon_election/connectivity rados supported-random-distro$/{ubuntu_latest}}",
                 ['/cephfs/home/user/src/git.ceph.com_ceph-c_aclamk-testing/qa/suites/rados/singleton-nomsgr/all/health-warnings.yaml',
                  '/cephfs/home/user/src/git.ceph.com_ceph-c_aclamk-testing/qa/suites/rados/singleton-nomsgr/mon_election/connectivity.yaml',
                  '/cephfs/home/user/src/git.ceph.com_ceph-c_aclamk-testing/qa/suites/rados/singleton-nomsgr/rados.yaml',
                  '/cephfs/home/user/src/git.ceph.com_ceph-c_aclamk-testing/qa/suites/rados/singleton-nomsgr/supported-random-distro$/ubuntu_latest.yaml'])
            ]
        ],
        [
            "rados/cephadm/osds/{0-distro/centos_9.stream_runc 0-nvme-loop 1-start 2-ops/rm-zap-flag},"
            "rados/standalone/{supported-random-distro$/{ubuntu_latest} workloads/scrub}",
            "/cephfs/home/user/src/git.ceph.com_ceph-c_squid/qa/suites",
            [
                ("rados/cephadm/osds/{0-distro/centos_9.stream_runc 0-nvme-loop 1-start 2-ops/rm-zap-flag}",
                 ['/cephfs/home/user/src/git.ceph.com_ceph-c_squid/qa/suites/rados/cephadm/osds/0-distro/centos_9.stream_runc.yaml',
                  '/cephfs/home/user/src/git.ceph.com_ceph-c_squid/qa/suites/rados/cephadm/osds/0-nvme-loop.yaml',
                  '/cephfs/home/user/src/git.ceph.com_ceph-c_squid/qa/suites/rados/cephadm/osds/1-start.yaml',
                  '/cephfs/home/user/src/git.ceph.com_ceph-c_squid/qa/suites/rados/cephadm/osds/2-ops/rm-zap-flag.yaml'])
                ,
                ("rados/standalone/{supported-random-distro$/{ubuntu_latest} workloads/scrub}",
                 ['/cephfs/home/user/src/git.ceph.com_ceph-c_squid/qa/suites/rados/standalone/supported-random-distro$/ubuntu_latest.yaml',
                  '/cephfs/home/user/src/git.ceph.com_ceph-c_squid/qa/suites/rados/standalone/workloads/scrub.yaml'])
            ]
        ],
    ])
    def test_yamls_from_desc(self, desc, base, result):
        res = list(run.descr_to_yamls(desc, base))
        print(f"Result: {res}")
        assert res == result 

```

I took the example you provided, but it would be more general if you check other test for matrix and build_matrix...